experimental/dds/tree2/src/feature-libraries/modular-schema/typedSchema/schemaBuilder.ts

```diff
@@ -123,6 +128,96 @@ export class SchemaBuilder {
 		return this.object(name, t as TreeSchemaSpecification) as TreeSchema<Name, T>;
 	}
 
+	/**
+	 * Define (and add to this library) a {@link TreeSchema} for a struct node.
+	 *
+	 * The name must be unique among all node schema in the the document schema.
+	 */
+	public struct<Name extends string, T extends RestrictiveReadonlyRecord<string, FieldSchema>>(
+		name: Name,
+		t: T,
+	): TreeSchema<Name, { local: T }> {
+		const schema = new TreeSchema(this, name, { local: t });
+		this.addNodeSchema(schema);
+		return schema;
+	}
+
+	/**
+	 * Same as `struct` but with less type safety and works for recursive objects.
+	 * Reduced type safety is a side effect of a workaround for a TypeScript limitation.
+	 *
+	 * See note on RecursiveTreeSchema for details.
+	 */
+	public structRecursive<Name extends string, T>(
+		name: Name,
+		t: T,
+	): TreeSchema<Name, { local: T }> {
+		return this.struct(
+			name,
+			t as unknown as RestrictiveReadonlyRecord<string, FieldSchema>,
+		) as unknown as TreeSchema<Name, { local: T }>;
+	}
+
+	/**
+	 * Define (and add to this library) a {@link TreeSchema} for a map node.
+	 *
+	 * The name must be unique among all node schema in the the document schema.
+	 */
+	public map<Name extends string, T extends FieldSchema>(
+		name: Name,
+		t: T,
+	): TreeSchema<Name, { extraLocalFields: T }> {
+		const schema = new TreeSchema(this, name, { extraLocalFields: t });
```

Comment:
Are extraLocalFields still being used for anything else? If not, I vote we change the name.

Reply:
they will be deleted when this migration is done, so no need to do a rename.